### PR TITLE
{numlib,phys}[GCC/11.2.0] CLHEP v2.4.4.0, Geant4 v10.7.1

### PR DESCRIPTION
**Pull request by ItIsI-Orient**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by ItIsI-Orient** [comment]
Test report by @ItIsI-Orient
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3201.victini.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/d2d2df004a270a233b4fbb355f37f008 for a full test report.

File: easybuild/easyconfigs/g/Geant4/Geant4-10.7.1-GCC-11.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+github_account = 'Geant4'
+source_urls = [GITHUB_SOURCE]
+sources = [{'download_filename': 'v%(version)s.tar.gz', 'filename': SOURCELOWER_TAR_GZ}]
+checksums = ['a22800227713b46ba0b71bc674f6c937e5cc0cf17a6becf4a10d5ef90693a60b']
+
+builddependencies = [('CMake', '3.22.1')]
+dependencies = [
+    ('expat', '2.4.1'),
+    # recommended CLHEP version, see https://geant4-data.web.cern.ch/geant4-data/ReleaseNotes/ReleaseNotes4.10.6.html
```
Comment (boegel) [inline easybuild/easyconfigs/g/Geant4/Geant4-10.7.1-GCC-11.2.0.eb:19]: fix URL, should be https://geant4-data.web.cern.ch/geant4-data/ReleaseNotes/ReleaseNotes4.10.7.html

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
This may need some more work, to avoid using C++23 standard for GATE in #14869

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (2 easyconfigs in total)
node2677.swalot.os - Linux UNKNOWN UNKNOWN, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/766873e6c7dc0b4c886138609b0e65ee for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3523.doduo.os - Linux UNKNOWN UNKNOWN, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/31018dcd805d710a60cacc6facd78b47 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @ItIsI-Orient!

**Pull request merged**